policy/version.go

```diff
@@ -114,17 +114,53 @@ func Version_Expression_Factory(ver_string string) (*Version_Expression, error)
 		end_inclusive:   rightIncluded(expr),
 	}
 
+	// nomalize the versions in the expression
+	ve.recalc_expression()
```

Comment:
To normalize the version expression. There must be a reason. I cannot remember now. :-)

Reply:
That should not be needed. The Factory method already normalizes the version strings, so if there is a path where that doesnt happen, when we should fix it at that point. This happens at lines 99 and 106.

Reply:
recalc_expression() is exactly doing what you are saying which fixes the expression string.  Since it is needed in other places too.  I put is in a separate function.